mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -71,6 +71,22 @@ private[recommendation] trait ALSModelParams extends Params with HasPredictionCo
 
   /** @group getParam */
   def getItemCol: String = $(itemCol)
+
+  /**
+   * Param for strategy for dealing with unknown users or items at prediction time.
+   * Supported values:
+   * - "nan": prediction value for unknown ids will be NaN.
+   * - "drop": rows in the input DataFrame containing unknown ids will be dropped.
+   * Default: "nan".
+   * @group expertParam
+   */
+  val unknownStrategy = new Param[String](this, "unknownStrategy",
```

Comment:
Fair enough - I was trying to keep it brief. But it could be more clear.
How about "unknownIdPredictionStrategy"?
On Wed, 4 May 2016 at 17:23, Seth Hendrickson <notifications@github.com>
wrote:

> In mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala
> <https://github.com/apache/spark/pull/12896#discussion_r62059332>:
>
> > @@ -71,6 +71,22 @@ private[recommendation] trait ALSModelParams extends Params with HasPredictionCo
> >
> >    /** @group getParam */
> >    def getItemCol: String = $(itemCol)
> > +
> > +  /**
> > +   * Param for strategy for dealing with unknown users or items at prediction time.
> > +   * Supported values:
> > +   * - "nan": prediction value for unknown ids will be NaN.
> > +   * - "drop": rows in the input DataFrame containing unknown ids will be dropped.
> > +   * Default: "nan".
> > +   * @group expertParam
> > +   */
> > +  val unknownStrategy = new Param[String](this, "unknownStrategy",
>
> I don't like the name "unknownStrategy" - seems very ambiguous. I'm not
> sure what is best, but for example I think "unknownPredictionStrategy" or
> "newUserItemPredictionStrategy" are better. Just something that will be
> obvious what it is by the name.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/12896/files/fc437451a598221f0878b7a2e0b87d17572019cc#r62059332>
>